homeassistant/components/ovo_energy/__init__.py

```diff
@@ -40,7 +40,14 @@ async def async_update_data() -> OVODailyUsage:
         """Fetch data from OVO Energy."""
         now = datetime.utcnow()
         async with async_timeout.timeout(10):
-            return await client.get_daily_usage(now.strftime("%Y-%m"))
+            try:
+                await client.authenticate(
+                    entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD]
+                )
+                return await client.get_daily_usage(now.strftime("%Y-%m"))
+            except aiohttp.ClientError as exception:
+                _LOGGER.warning(exception)
+                return None
```

Comment:
In the future in another PR, we can start a reauth step in the config flow here.

Reply:
Good plan :smile_cat: